### DISCUSSION TITLE
importing config file while RustDesk is running doesn't work

### DISCUSSION THREAD

**Discussion by piterq** [rustdesk/rustdesk#6300] [General]
### Bug Description

I'm using different RustDesk servers, so I must change them often in configuration while the application is running.

### How to Reproduce

Copy custom rustdesk.toml and rustdesk2.toml files to c:\rustdesk_config directory, start RustDesk (with service running) and try to execute:
rustdesk.exe --import-config c:\rustdesk_config\rustdesk.toml

### Expected Behavior

I expect that server and key values in "Network" section in RustDesk configuration will be replaced with new ones. This doesn't happen. I've even tried to write batch file which stops RustDesk service, kills RustDesk.exe and replaces RustDesk2.toml file, but that doesn't help - as soon as I start RustDesk, contents of this file are reversed to original.

### Operating system(s) on local side and remote side

Windows 11 -> Windows 11

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

Screenshots are unnecessary.

**Comment by kproske** [comment]
Hello,

we have the same Problem.
We use the export funktion.
The export funktion generate a Hash-Code in the Clipboard.

Paste the Hash-Code in an Editor an use `rustdesk.exe --config <HASH-CODE>`

We use this Option to rollout rustdesk with OPSI, without the Pro Version.

I hope this is helpfull.

Kind Regards.

**Reply by Unknown78** [reply]
Hello @kproske,

Where is the export function? And where to generate the hash code?
I'm not a pro version user, just an open source one.

**Reply by BumbleBeeUser** [reply]
The export function is located on your Rustdesk server.
An CLI command.
We use a Docker image, but I don't remember exactly where and how that worked.
Perhaps you could ask an AI, or check the Rustdesk wiki again.

Sorry, it's been a while since I did that.

**Reply by Unknown78** [reply]
@BumbleBeeUser 

https://github.com/search?q=repo%3Arustdesk%2Frustdesk-server%20export&type=code
There is no keyword of export that's related to it in the open source version (OSS) of rustdesk server.

Are you sure it's not a pro version feature?

**Comment by BumbleBeeUser** [comment]
I´m using the non Pro Version.

**Reply by Unknown78** [reply]
I couldn't find it :(

**Reply by BumbleBeeUser** [reply]
See above:
"The export function is located on your Rustdesk server.
An CLI command.
We use a Docker image, but I don't remember exactly where and how that worked.
Perhaps you could ask an AI, or check the Rustdesk wiki again."

**Reply by Unknown78** [reply]
Ok i get it.

It's the icons on top right of the `Client`'s `ID/Relay Server` GUI settings.
There are no texts, just icons, that's why I miss it.

The export/import has GUI.
But only the import has the CLI of `rustdesk --config <config‑string>`.

I've found the export GUI but not the export CLI.

---

This might be the import code, but to generate the export code... reverse engineering is tiring...

https://github.com/rustdesk/rustdesk/blob/317639169359936f7f9f85ef445ec9774218772d/src/core_main.rs#L418-L442
